tests/document/instance.py

```diff
@@ -826,6 +828,27 @@ def test_modify_update(self):
 
         self.assertDbEqual([dict(other_doc.to_mongo()), dict(doc.to_mongo())])
 
+    @needs_mongodb_v26
+    def test_modity_push_position(self):
```

Comment:
What is the reason for all of the complexity in this test? Aren't you simply trying to assert that the `modify` method with a positional push works? If so, this is much simpler and more explicit about its intent:

```py
def test_modify_with_positional_push(self):
    class BlogPost(Document):
        tags = ListField(StringField())

    post = BlogPost.objects.create(tags=['python'])
    self.assertEqual(post.tags, ['python'])
    post.modify(push__tags__0=['code', 'mongo'])
    self.assertEqual(post.tags, ['code', 'mongo', 'python'])
    
    # Assert same order of the list items is maintained in the db
    self.assertEqual(
        BlogPost._get_collection().find_one({'_id': post.pk})['tags'],
        ['code', 'mongo', 'python']
    )
```